File: app/src/main/java/com/base/base_app/data/model/workperiod/GetAllWorkperiods.java
package com.base.base_app.data.model.workperiod;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

@Entity(tableName = "get_work_period")
public class GetAllWorkperiods {
    @SerializedName("Id")
    @Expose
    @NonNull
    @PrimaryKey
    private Integer id;
    @SerializedName("Workperiods")
    @Expose
    private List<Workperiod> workperiods = null;
    @SerializedName("CurrentWorkperiod")
    @Expose
    private Workperiod currentWorkperiod;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public List<Workperiod> getWorkperiods() {
        return workperiods;
    }

    public void setWorkperiods(List<Workperiod> workperiods) {
        this.workperiods = workperiods;
    }

    public Workperiod getCurrentWorkperiod() {
        return currentWorkperiod;
    }

    public void setCurrentWorkperiod(Workperiod currentWorkperiod) {
        this.currentWorkperiod = currentWorkperiod;
    }
}

File: app/src/main/java/com/base/base_app/data/model/workperiod/GetAllWorkperiods.kt
package com.base.base_app.data.model.workperiod

import androidx.room.Entity
import com.google.gson.annotations.SerializedName
import com.google.gson.annotations.Expose
import androidx.room.PrimaryKey
import com.base.base_app.data.model.workperiod.Workperiod

@Entity(tableName = "get_work_period")
data class GetAllWorkperiods (
    @SerializedName("Id")
    @Expose
    @PrimaryKey
    var id: Int ,

    @SerializedName("Workperiods")
    @Expose
    var workperiods: List<Workperiod>? = null,

    @SerializedName("CurrentWorkperiod")
    @Expose
    var currentWorkperiod: Workperiod? = null
)